streaming/src/test/scala/org/apache/spark/streaming/StreamingContextSuite.scala

```diff
@@ -297,6 +299,23 @@ class StreamingContextSuite extends SparkFunSuite with BeforeAndAfter with Timeo
     Thread.sleep(100)
   }
 
+  test("registering and de-registering of streamingSource") {
+    val conf = new SparkConf().setMaster(master).setAppName(appName)
+    ssc = new StreamingContext(conf, batchDuration)
+    addInputStream(ssc).register()
+
+    ssc.start()
+    assert(ssc.getState() === StreamingContextState.INITIALIZED)
+    assert(StreamingContextSuite.sources.get(StreamingContextSuite.streamingSource)!= "null")
```

Comment:
I dont get this statement here. You are calling StreamingContext.sources.get() on an object of type `java.lang.reflect.Field` (that is, StreamingContextSuite.streamingSource) rather than the actual StreamingContext object `ssc` above. And then you are comparing that it is not the string "null"? I am not sure what is being tested here.